src/vivarium_public_health/results/disability.py

```diff
@@ -64,19 +68,54 @@ def __init__(self) -> None:
     def setup(self, builder: Builder) -> None:
         self.step_size = pd.Timedelta(days=builder.configuration.time.step_size)
         self.disability_weight = self.get_disability_weight_pipeline(builder)
-        self.causes_of_disease = [
-            cause
-            for cause in builder.components.get_components_by_type(
-                tuple(self.disease_classes)
+        self.set_causes_of_disability(builder)
+
+    def set_causes_of_disability(self, builder: Builder) -> None:
+        """Set the causes of disability to be observed by removing any excluded
+        via the model spec from the list of all disability class causes. We implement
+        exclusions here because disabilities are unique in that they are not
+        registered stratifications and so cannot be excluded during the stratification
+        call like other categories.
+        """
+        causes_of_disability = list(
+            builder.components.get_components_by_type(tuple(self.disability_classes))
```

Comment:
Hmm, this was my attempt to remove unnecessary list comprehension since whatever was returned could just be cast to `list`. But I think you're right, the actual return of `get_components_by_type` is already a list so no need to re-cast.

As for the tuple, it is indeed type-hinted that way. A quick look tells me it could take the `Union[Tuple, list]` but I'm hesitant to start down that rabbit hole unless you think it's worth it.

Reply:
ok, was easy enough to allow for lists to be passed in and then coerce later to tuple. (tuple is needed down the line for an isinstance call)